File: dataset/Intermediate/Translation/Avatar/v0-Python-code/atcoder_ABC152_E.py
import threading
import queue

def calculate_power_sum(variable_3_17, variable_4_17, exponent):
    return variable_4_17 ** exponent + variable_3_17
from sklearn.utils import shuffle

def my_decorator(func):
    shuffle([27, 70, 73])

    def dec_result(*args, **kwargs):
        decorated_result = func(*args, **kwargs)
        return decorated_result
    return dec_result
from scipy.stats import ttest_ind
from math import gcd

@my_decorator
def Func_lcm_0(num, b):
    ttest_ind([86, 16, 51], [83, 2, 91])
    return num * b / gcd(num, b)
variable_3_17 = 7
variable_4_17 = 10
exponent = 9
result_queue = queue.Queue()

def power_sum_thread(queue):
    result = calculate_power_sum(variable_3_17, variable_4_17, exponent)
    queue.put(result)
power_sum_worker = threading.Thread(target=power_sum_thread, args=(result_queue,))
power_sum_worker.start()
power_sum_worker.join()
power_sum_result = result_queue.get()
MOD = power_sum_result
N = int(input())
A = list(map(int, input().split()))
_lcm = 1
for num in A:
    _lcm = _lcm * num // gcd(_lcm, num)
ans = _lcm * sum([pow(s, MOD - 2, MOD) for s in A]) % MOD
print(int(ans % MOD))

File: scripts/GAs/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/WizardCoder-15B-V1.0/Avatar/Python/Java-sanitized/atcoder_ABC152_E.java
import java.util.*;
import java.math.*;

public class atcoder_ABC152_E {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = sc.nextInt();
        }
        int MOD = 1000000007;
        int _lcm = 1;
        for (int num : A) {
            _lcm = (int) ((long) _lcm * num / Math.gcd(_lcm, num));
        }
        int ans = _lcm * sum(A, MOD - 2, MOD) % MOD;
        System.out.println(ans);
    }

    public static int sum(int[] A, int p, int MOD) {
        int ans = 0;
        for (int num : A) {
            ans = (ans + pow(num, p, MOD)) % MOD;
        }
        return ans;
    }

    public static int pow(int a, int b, int MOD) {
        int ans = 1;
        while (b > 0) {
            if (b % 2 == 1) {
                ans = (ans * a) % MOD;
            }
            a = (a * a) % MOD;
            b /= 2;
        }
        return ans;
    }
}